101-application-gateway-rewrite/azuredeploy.json

```diff
@@ -0,0 +1,312 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "addressPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/16",
+      "metadata": {
+        "description": "Address prefix for the Virtual Network"
+      }
+    },
+    "subnetPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/28",
+      "metadata": {
+        "description": "Subnet prefix"
+      }
+    },
+    "skuName": {
+      "type": "string",
+      "defaultValue": "Standard_Medium",
+      "metadata": {
+        "description": "Sku Name"
+      }
+    },
+    "backendIpAddress1": {
+      "type": "string",
+      "metadata": {
+        "description": "IP Address for Backend Server 1"
+      }
+    },
+    "backendIpAddress2": {
+      "type": "string",
+      "metadata": {
+        "description": "IP Address for Backend Server 2"
+      }
+    },
+    "MSIVersion": {
+      "type": "string",
+      "defaultValue": "2015-08-31-PREVIEW",
+      "metadata": {
+        "description": "API version for msi extension"
+      }
+    },
+    "MSIPort": {
+      "type": "int",
+      "defaultValue": 50342,
+      "metadata": {
+        "description": "Default port for msi extension"
+      }
+    },
+    "KeyVaultVersion": {
+      "type": "string",
+      "defaultValue": "2018-02-14-preview",
+      "metadata": {
+        "description": "API version for key vault resources and access policy resources"
+      }
+    },
+    "Secrets": {
+      "type": "array",
+      "metadata": {
+        "description": "An array of json objects like this : {'name':name, 'value':value}"
+      }
+    }
+  },
+  "variables": {
+    "applicationGatewayName": "[concat(resourceGroup().name, '-appgw')]",
+    "idName": "identity1",
+    "keyVaultName": "resource-kv",
+    "publicIPAddressName": "[concat(resourceGroup().name, '-publicIp')]",
+    "virtualNetworkName": "[concat(resourceGroup().name, '-vnet')]",
+    "subnetName": "subnet1",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+    "publicIPRef": "[resourceId('Microsoft.Network/publicIPAddresses',variables('publicIPAddressName'))]",
+    "applicationGatewayID": "[resourceId('Microsoft.Network/applicationGateways',variables('applicationGatewayName'))]",
+    "apiVersion": "2018-10-01",
+    "identityID": "[resourceId('Microsoft.ManagedIdentity/userAssignedIdentities',variables('idName'))]",
+    "keyVaultID": "[resourceId('Microsoft.KeyVault/vaults',variables('keyVaultName'))]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.ManagedIdentity/userAssignedIdentities",
+      "name": "[variables('idName')]",
+      "apiVersion": "2015-08-31-PREVIEW",
+      "location": "[resourceGroup().location]"
```

Comment:
Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

Reply:
Added.

Reply:
you need to use the location parameter for all location properties

Reply:
done.